### DISCUSSION TITLE
Compilation errors using VS2022 regarding headers

### DISCUSSION THREAD

**Discussion by NicusorN5** [NVIDIA-Omniverse/PhysX#215] [Q&A] [answered]
Hi, I have downloaded `physx 5.3.0` using `vcpkg` and I'm trying to instantiate a scene. 

I have issues when attempting to compile with missing headers:
```
C1083	Cannot open include file: 'geometry/PxCustomGeometry.h': No such file or directory
```
when including `<physx/PxScene.h>`. I have tried including other headers, and only `<physx/PxScene.h` seems to work. I don't expect this to be intended, so how can I fix this?

My current source looks like this currently:
```cpp
#pragma once
#include <physx/PxScene.h>

class PhysicsEngineAllocator : public physx::PxAllocatorCallback
{
	// Inherited via PxAllocatorCallback
	void* allocate(size_t size, const char* typeName, const char* filename, int line) override; //call to std::malloc
	void deallocate(void* ptr) override; //call to std::free
};
```

Any help is appreciated, thanks.

**Comment by ayoub-belarbi** [comment]
Hello @NicusorN5,
Indeed it's not intended like that.
The fact that it was able to find PxScene.h (in physx/include) and not PxCustomGeometry.h (in physx/include/geometry/) suggests a possible issue with how subdirectories are being handled in the include path settings.
Can you share your cmake config? How are you using PhysX in your project? Did you try adding  physx/include/geometry/ to the include path?

**Reply by NicusorN5** [reply]
Hi, I'm only using a VS2022 solution. It is entirely necessary to use CMake?

I'm only using VS2022 with vcpkg integrated.

**Reply by ayoub-belarbi** [reply, marked as answer]
Ok, although we don't officially maintain or support any vcpkg package, we did try to reproduce this, it seems that the issue is related to: https://github.com/microsoft/vcpkg/issues/8071#issuecomment-528778328
So in a nutshell, using:
`#include "geometry/PxCustomGeometry.h"` instead of `#include <geometry/PxCustomGeometry.h>` should fix the issue.
If it doesn't, I'm afraid that we can't really help with these issues. We can't support anything which is community driven and not provided by this repo.
Thanks

**Reply by NicusorN5** [reply]
Thanks, that seemed to work.
